app/src/main/java/chat/rocket/android/authentication/infraestructure/SharedPreferencesMultiServerTokenRepository.kt

```diff
@@ -0,0 +1,33 @@
+package chat.rocket.android.authentication.infraestructure
+
+import chat.rocket.android.authentication.domain.model.TokenModel
+import chat.rocket.android.dagger.scope.PerActivity
+import chat.rocket.android.infrastructure.LocalRepository
+import chat.rocket.android.server.domain.MultiServerTokenRepository
+import com.squareup.moshi.Moshi
+
+@PerActivity
+class SharedPreferencesMultiServerTokenRepository(private val repository: LocalRepository,
+                                                  private val moshi: Moshi
+) : MultiServerTokenRepository {
+
+    override fun get(server: String): TokenModel? {
+        val token = repository.get("$TOKEN_KEY$server")
+        val adapter = moshi.adapter<TokenModel>(TokenModel::class.java)
+
+        token?.let {
+            return adapter.fromJson(token)
+        }
+
+        return null
+    }
+
+    override fun save(server: String, token: TokenModel) {
+        val adapter = moshi.adapter<TokenModel>(TokenModel::class.java)
+
+        repository.save("$TOKEN_KEY$server", adapter.toJson(token))
+    }
+
+}
+
+const val TOKEN_KEY = "token_"
```

Comment:
This is related to the "customer's taste", but once we are defining the class properties on the top we could put our `const` on the top of the class too.

A little of history: usually a developer first looks at class behaviour's implementation (member functions or methods). In some projects we can see the member variables at the bottom of the class and I really like this approach (but this is unusual on Android app development) - making the class easy to read (like you did here).

Reply:
Constants should be self descriptive, there is no need for them to be at the top of the file. And moving to a companion object I'm following the kotlin coding convetions doc.

Reply:
A good example of putting the `const` at the top is here (when passing extras - not this case, just for reference): https://github.com/RocketChat/Rocket.Chat.Android/blob/develop-2.x/app/src/main/java/chat/rocket/android/chatroom/ui/ChatRoomActivity.kt

Reply:
Oh sorry, the article that I would like to share is: https://medium.com/@passsy/starting-activities-with-kotlin-my-journey-8b7307f1e460

This is also another nice article by the GDE (for Flutter :P)  Pascal @passsy Welsch